synapse/lib/types.py

```diff
@@ -309,12 +311,25 @@ def norm(self, valu):
         if self.opts['lower']:
             norm = norm.lower()
 
+        if self.opts['strip']:
+            norm = norm.strip()
+
         if self.regex is not None:
             if self.regex.match(norm) is None:
                 raise s_exc.BadTypeValu(name=self.name, valu=valu, mesg='regex does not match')
 
         return norm, {}
 
+    def liftPropEq(self, xact, fenc, penc, valu):
```

Comment:
the whole prefix concept can go away now.  we have ^= for str type in my branch and that can be the de-facto way to do that to strings ( or things we want to behave like them )